nurseconnect/views.py

```
@@ -228,11 +229,26 @@ def post(self, request, *args, **kwargs):
                     self.request.user.save()
                     return HttpResponseRedirect(reverse("view_my_profile"))
                 else:
-                    messages.success(
+                    profile_password_change_form.add_error(
+                        "old_password",
+                        ValidationError("The old password is incorrect.")
+                    )
+                    settings_form = forms.EditProfileForm(
+                        prefix="settings_form", user=self.request.user
+                    )
+                    profile_password_change_form.change_field_enabled_state(
+                        False
+                    )
+                    return render(
                         request,
-                        "The old password is incorrect."
+                        self.template_name,
+                        context={
+                            "edit": "edit-password",
+                            "settings_form": settings_form,
+                            "profile_password_change_form":
+                                profile_password_change_form
+                        }
                     )
-                    return HttpResponseRedirect(reverse("view_my_profile"))
```

why are we doing away with this?

---

When you do a `HttpResponseRedirect` it instantiates a new form and discards the errors that I just added to the form